File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    let counter=0,maxi=0;
   
    s.split('').forEach(char => {
        if (char === '(') {
            counter++;
            maxi = Math.max(maxi, counter);
        } else if (char === ')') {
            counter--;
        }
    });
    return maxi
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int counter=0,maxcounter=0;
        for(char it:s){
                if(it=='(')
                counter++;
                if(it==')')
                counter--;
                maxcounter=max(maxcounter,counter);
        }
        return maxcounter;
    }
};